src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -356,7 +321,7 @@ public override Stream GetStream()
         /// </summary>
         public override void Abort()
         {
-            _client.CancelPendingRequests();
+            _abortRequestCancellationTokenSource.Cancel();

```

Comment:
Did you add tests to verify this?

Reply:
Yes I did, see: https://github.com/OData/odata.net/pull/2883/files#diff-b4128975c89acf84c4f440fc4c0b2af782a08e0fb8955d0195ed344bab8562a8R115